habitat/tasks/rearrange/rearrange_task.py

```diff
@@ -74,6 +95,10 @@ def __init__(self, *args, sim, dataset=None, **kwargs) -> None:
         else:
             self._robot_pos_start = None
 
+        if len(self._sim.robots_mgr) > 1:
+            # Duplicate sensors that handle robots. One for each robot.
+            self._duplicate_sensor_suite(self.sensor_suite, False)
```

Comment:
Will this modification work for agents with different sensor suites. The config supports that, but the code doesn't as I understand it.

Reply:
It does not look like the config supports it either, there is only one sensor list in the config and they do not specify the agent they correspond to.

Reply:
@vincentpierre There are actually two sensor lists in the config. For example: https://github.com/facebookresearch/habitat-lab/blob/380ac0a7d8c4ead1532f109b15d329473212eae9/configs/tasks/rearrange/benchmark/idle_single_camera.yaml#L51 It's annoying because Habitat_Baselines has its own sensor list which overrides which sensors are enabled, but AGENT_0, AGENT_1, etc also have their own sensor lists for this purpose.

Reply:
This is only for visual sensors and it seems that even the resolution is defined for both agents at the same time. If you look at vector observations, they are defined for both agents at the same time. It is bothering me that `SENSORS: ['HEAD_RGB_SENSOR'] ` appears twice, but the sensors are copy pasted between agents. Is Agent_0 the ground truth of Agent_1? What happens if this line is different for both agents, should it raise an error or will the agents get a different sensor? 🤔

Reply:
The agents should get different sensors (each defined individually in the Sensor List). Think like a key lookup into the map defined by the sensor lsit.